dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLitmusC.java

```diff
@@ -252,7 +283,10 @@ public Expression visitReAtomicFetchOp(LitmusCParser.ReAtomicFetchOpContext ctx)
     public Expression visitC11AtomicOp(LitmusCParser.C11AtomicOpContext ctx) {
         Register register = getReturnRegister(true);
         Expression value = returnExpressionOrOne(ctx.value);
-        Event event = EventFactory.Atomic.newFetchOp(register, getAddress(ctx.address), value, ctx.op, ctx.c11Mo().mo);
+        Expression address = getAddress(ctx.address);
+        Event event = EventFactory.Atomic.newFetchOp(register, address, value, ctx.op, ctx.c11Mo().mo);
+        addMemoryObjectTags(event, address);
+        addScopeTag(event, ctx.openCLScope());
```

Comment:
Shouldn't this be conditional on the instruction actually having a scope? I think right now this might be `null`

Reply:
In addScopeTag() it checks if the scope context is `null`, if it's `null` and `isOpenCL`, we add the default scope.